stories/patterns/ux-patterns/confirmation-modal-dialog/docs/examples/confirmation-modal-example.tsx

```diff
@@ -0,0 +1,149 @@
+import '@semcore/utils/lib/themes/default.css';
+import { useForm, SubmitHandler } from 'react-hook-form';
+
+const warningBlockStyles = {
+  background: 'var(--intergalactic-bg-secondary-critical)',
+  border: '1px solid var(--intergalactic-border-critical)',
+  borderRadius: 'var(--intergalactic-surface-rounded)',
+  marginBottom: '24px',
+  padding: '16px',
+};
+
+type FormValues = {
+  project: string;
+};
+
+const fieldName: keyof FormValues = 'project';
+
+const Demo = () => {
+  const { register, trigger, handleSubmit, reset, resetField, getValues, formState } =
+    useForm<FormValues>({
+      mode: 'onBlur',
+      defaultValues: { project: '' },
+    });
+
+  const [visible, setVisible] = React.useState(false);
+  const handleOpen = React.useCallback(() => setVisible(true), []);
+  const handleClose = React.useCallback(() => {
+    setVisible(false);
+    reset({ project: '' }, { keepIsSubmitted: false, keepTouched: false });
+  }, []);
+
+  const { errors, isSubmitted, touchedFields } = formState;
+  const [focusedFieldName, setFocusedFieldName] = React.useState('');
+  const onSubmit: SubmitHandler<FormValues> = handleClose;
+
+  const hasError = () => {
+    const error = errors[fieldName];
+
+    if (error && error.type === 'projectRequired' && !isSubmitted) {
+      return false;
+    }
+
+    return Boolean(error);
+  };
+  
+  const showErrorTooltip = (): boolean => {
+    const isActive = focusedFieldName === fieldName;
+    return hasError() && isActive;
+  };
+
+  const { onChange, ...restField } = register(fieldName, {
+    validate: {
+      projectRequired: (v) => {
+        return Boolean(v) || 'Please enter correct project name';
+      },
+      projectName: (v) => {
+        if (!v) {
+          return true;
+        }
+
+        return /^test$/i.test(v) || 'Please enter correct project name';
+      },
+    },
+    onBlur: () => setFocusedFieldName(''),
+  });
+
+  const field = {
+    onChange: (_v: string, e: React.SyntheticEvent) => {
+      // important: keep call order, otherwise validation breaks
+      onChange(e);
+      hasError() && trigger();
+    },
+    ...restField,
+  };
+
+  return (
+    <>
+      <Button onClick={handleOpen}>Open modal</Button>
+      <Modal visible={visible} onClose={handleClose} w={536}>
+        <Modal.Title mb={4}>Delete project?</Modal.Title>
+        <Text size={300} mb={4} tag='p'>
+          This will <Text tag='strong'>delete</Text> the following campaigns set up for{' '}
+          <Text tag='strong'>test.com</Text> with all their data:
+        </Text>
+        <List size={300} mb={4}>
+          <List.Item>Position Tracking</List.Item>
+          <List.Item>Social Media Poster</List.Item>
+          <List.Item>Backlink Audit</List.Item>
+          <List.Item>Content Analyzer</List.Item>
+        </List>
+
+        <form noValidate onSubmit={handleSubmit(onSubmit)}>
+          <Flex style={warningBlockStyles} tag='label' direction='column' htmlFor='project'>
```

Comment:
```suggestion
          <Flex style={warningBlockStyles} tag='label' direction='column' htmlFor={fieldName}>
```
maybe better use the variable?